.github/workflows/record-release-interval/action.yml

```
@@ -35,22 +35,31 @@ runs:
         # This is all one pipeline. Edit with care.
         # [
             # Get the last 1000 events from datadog
-            curl "https://api.datadoghq.com/api/v1/events?start=${starttime}&end=${now}&tags=env:prod,deploy:true&unaggregated=true" \
+            curl "https://api.datadoghq.com/api/v1/events?start=${starttime}&end=${now}&tags=env:${{ inputs.environment }},deploy:true&unaggregated=true" \
                 -H "Content-Type: text/json" \
                 -H "DD-APPLICATION-KEY: ${{ inputs.datadog_app_key }}" \
                 -H "DD-API-KEY: ${{ inputs.datadog_api_key }}" | \
 
             # Filter the events down to just content releases and put the most recent timestamp into last_deploy.txt
-            jq '[.events[] | select(.title=="Content release (prod)")] | first | .date_happened' > last_deploy.txt
+            jq '[.events[] | select(.title=="Content release (${{ inputs.environment }})")] | first | .date_happened' > last_deploy.txt
         # ]
 
-        time_since_last_deploy="$(expr ${now} - $(cat last_deploy.txt))"
-
         # There's probably a better way to handle missing event timestamps, but
         # this is relatively simple and will work for now.
         last_deploy=$(cat last_deploy.txt || echo $CURRENT_TIME)
         echo "LAST_DEPLOY=${last_deploy})" >> $GITHUB_ENV
-        echo "TIME_SINCE_LAST_DEPLOY=$(expr ${now} - ${last_deploy})" >> $GITHUB_ENV
+
+        time_since_last_deploy=$(expr ${now} - ${last_deploy})
+        echo "TIME_SINCE_LAST_DEPLOY=${time_since_last_deploy}" >> $GITHUB_ENV
+
+        # Subtract time in 60 minute intervals until we're under 60 minutes.
+        # This should get us pretty close to actual business hours (removes
+        # nights, weekends, and other long hiatuses).
+        while [ $time_since_last_deploy -ge 3600 ]; do
+            time_since_last_deploy=$(expr ${time_since_last_deploy} - 3600)
+        done
+
+        echo "BIZTIME_SINCE_LAST_DEPLOY=${time_since_last_deploy}" >> $GITHUB_ENV
```

Couldn't this just be:
```suggestion
        echo "BIZTIME_SINCE_LAST_DEPLOY=$(expr ${time_since_last_deploy} % 3600)" >> $GITHUB_ENV
```

---

Sure could! Good catch. I had some other stuff in there and ended up repurposing the loop, but the modulo operator is definitely a better way to handle this.

---

Yeah, I figured it was one of:

a) left over code from repurposing
b) BSD expr interprets `%` as a bitwise NAND